### PR TITLE
added api get of codes without auth

### PR DESCRIPTION
**Pull request by mattlindsey**
https://www.pivotaltracker.com/story/show/134506351

Added api get of codes which will be /api/user/:id (since eventually it should return the user and it's codes).
Authentication has not been added yet, and currently only 1 'code' is returned.  Serialization of multiple codes should be added later.

### CODE REVIEW

**Comment by mattlindsey** [comment]
Note that the description of this PR is a little confusing, since the actual code that returns the vendor 'code' was in a prior PR and is in this file:
https://github.com/strawberrycanyon/redeemify/pull/43/files#diff-642333ee749a27d79361ea336d502db4

File: features/step_definitions/user_steps.rb [hunk 1]
```diff
@@ -205,3 +206,16 @@ def with_scope(locator)
 Then(/^I have updated the vendor home$/) do
   pending # Write code here that turns the phrase above into concrete actions
 end
+
```
Comment (mattlindsey) [inline features/step_definitions/user_steps.rb]: The 2 steps below are not used in any cucumber test (yet).  So maybe I should remove them?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,2 +1,11 @@
 class UsersController < ApplicationController
+
+  # This method is here temporarily so that the cucumber api_get.feature passes
+  # both when running cucumber and on the CI environment.
+  # Otherwise we would need somthing like 'prax' running.
+  def index
+    render json: {
+         code: '12345'
+    }.to_json
+  end
```
Comment (mattlindsey) [inline app/controllers/users_controller.rb]: I'm working on a solution to this problem using Rack::Proxy, so don't review this yet!!!

File: features/step_definitions/user_steps.rb [hunk 3]
```diff
@@ -145,6 +146,13 @@ def with_scope(locator)
   vendor.email = 'test@gmail.com'
   vendor.cashValue = '1'
   vendor.save
+  user = User.new
```
Comment (tansaku) [inline features/step_definitions/user_steps.rb]: could use factory girl, but do that in a separate ticket I think

File: features/step_definitions/user_steps.rb [hunk 4]
```diff
@@ -205,3 +213,12 @@ def with_scope(locator)
 Then(/^I have updated the vendor home$/) do
   pending # Write code here that turns the phrase above into concrete actions
 end
+
+Then /the response is a list containing three codes/ do
+  body = JSON.parse(last_response.body)
+  c = 0
+  body['code'].each do |code|
+    c = c + 1
+  end
```
Comment (tansaku) [inline features/step_definitions/user_steps.rb]: prefer `inject` e.g.

```rb
body['code'].inject(0){|memo, _code| memo = memo + 1}
```
but also we are not even refering to `c` here, so maybe this is unused code that can be deleted?

**Comment by tansaku** [comment]
Great work here @mattlindsey - I'm a little disturbed that the build is broken - have we resolved all the travis issues?  Is this specific to this PR?  You said something about failing cukes.

We need to get to a trustworthy green build soon to be able to use the CI effectively

File: app/controllers/api/v1/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+class Api::V1::UsersController < ApplicationController
+  respond_to :json
+
+
```
Comment (tansaku) [inline app/controllers/api/v1/users_controller.rb]: remove extra dead line

File: config/application.rb [hunk 6]
```diff
@@ -1,6 +1,19 @@
...
 
 require 'rails/all'
+require 'rack'
+require 'rack-proxy'
+
+class Proxy < Rack::Proxy
+   def perform_request(env)
+     request = Rack::Request.new(env)
+     if request.path =~ %r{^/api}
+       #do nothing
```
Comment (tansaku) [inline config/application.rb]: if this is do nothing why don't we swap to an unless?

**Pull request merged**